components/Ink/src/MDCInkView.m

```diff
@@ -69,59 +60,13 @@ - (instancetype)initWithFrame:(CGRect)frame {
 - (instancetype)initWithCoder:(NSCoder *)aDecoder {
   self = [super initWithCoder:aDecoder];
   if (self) {
-    if ([aDecoder containsValueForKey:MDCInkViewAnimationDelegateKey]) {
-      _animationDelegate = [aDecoder decodeObjectForKey:MDCInkViewAnimationDelegateKey];
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewMaskLayerKey]) {
-      _maskLayer = [aDecoder decodeObjectForKey:MDCInkViewMaskLayerKey];
-      _maskLayer.delegate = self;
-    } else {
-      _maskLayer = [CAShapeLayer layer];
-      _maskLayer.delegate = self;
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewUsesLegacyInkRippleKey]) {
-      _usesLegacyInkRipple = [aDecoder decodeBoolForKey:MDCInkViewUsesLegacyInkRippleKey];
-    } else {
-      _usesLegacyInkRipple = YES;
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewInkStyleKey]) {
-      self.inkStyle = [aDecoder decodeIntegerForKey:MDCInkViewInkStyleKey];
-    }
-
-    // The following are derived properties, but `layer` may not have been encoded
-    if ([aDecoder containsValueForKey:MDCInkViewUsesCustomInkCenterKey]) {
-      self.usesCustomInkCenter = [aDecoder decodeBoolForKey:MDCInkViewUsesCustomInkCenterKey];
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewCustomInkCenterKey]) {
-      self.customInkCenter = [aDecoder decodeCGPointForKey:MDCInkViewCustomInkCenterKey];
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewMaxRippleRadiusKey]) {
-      self.maxRippleRadius = (CGFloat)[aDecoder decodeDoubleForKey:MDCInkViewMaxRippleRadiusKey];
-    }
-    if ([aDecoder containsValueForKey:MDCInkViewInkColorKey]) {
-      self.inkColor = [aDecoder decodeObjectOfClass:[UIColor class] forKey:MDCInkViewInkColorKey];
-    }
+    _maskLayer = [CAShapeLayer layer];
```

Comment:
do you think that maybe instead we could use `commonMDCInkViewInit`?

Reply:
Am not sure - I want to avoid introducing new behavior in this PR, but we could explore cleaning this up in a follow-up PR.

Reply:
https://github.com/material-components/material-components-ios/issues/4627